src/Monolog/CompatibilityProcessingHandlerTrait.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Monolog;
+
+use Monolog\Level;
+use Monolog\Logger;
+use Monolog\LogRecord;
+use Sentry\Severity;
+
+if (Logger::API >= 3) {
+    /**
+     * Logic which is used if monolog >= 3 is installed.
+     *
+     * @internal
+     */
+    trait CompatibilityProcessingHandlerTrait
+    {
+        /**
+         * @param array<string, mixed>|LogRecord $record
+         */
+        abstract protected function doWrite($record): void;
+
+        /**
+         * {@inheritdoc}
+         */
+        protected function write(LogRecord $record): void
+        {
+            $this->doWrite($record);
+        }
+
+        /**
+         * Translates the Monolog level into the Sentry severity.
+         */
+        private static function getSeverityFromLevel(int $level): Severity
+        {
+            $level = Level::from($level);
```

Comment:
I tried to use `Level::Debug->value` in the `switch` statement but this broke the tests in PHP < 8.0.

Reply:
Makes definitely sense: we could not use anyway the PHP syntax for a newer version in a file that is also used in a previous version